nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,449 @@
+        ----------
+        sprite : numpy array with the sprite
+    """
+
+    # Get cmap
+    if isinstance(cm, str):
+        cmap = plt.cm.get_cmap(cmap)
+
+    img = check_niimg_3d(img, dtype='auto')
+
+    # resample to isotropic voxel with standard orientation
+    if resample:
+        img = _resample_to_self(img, interpolation)
+
+    # Read data
+    data = _safe_get_data(img, ensure_finite=True)
+    if np.isnan(np.sum(data)):
+        data = np.nan_to_num(data)
+
+    # Deal with automatic settings of plot parameters
+    if threshold == 'auto':
+        # Threshold epsilon below a percentile value, to be sure that some
+        # voxels pass the threshold
+        threshold = fast_abs_percentile(data) - 1e-5
+
+    # threshold
+    threshold = float(threshold) if threshold is not None else None
+
+    # Get vmin vmax
+    show_nan_msg = False
+    if vmax is not None and np.isnan(vmax):
+        vmax = None
+        show_nan_msg = True
+    if vmin is not None and np.isnan(vmin):
+        vmin = None
+        show_nan_msg = True
+    if show_nan_msg:
+        nan_msg = ('NaN is not permitted for the vmax and vmin arguments.\n'
+                   'Tip: Use np.nanmax() instead of np.max().')
+        warnings.warn(nan_msg)
+
+    if vmax is None:
+        vmax = np.nanmax(data)
+    if vmin is None:
+        vmin = np.nanmin(data)
+
+    # Create sprite
+    sprite = _data2sprite(data)
+
+    # Mask sprite
+    if threshold is not None:
+        if threshold == 0:
+            sprite = np.ma.masked_equal(sprite, 0, copy=False)
+        else:
+            sprite = np.ma.masked_inside(sprite, -threshold, threshold,
+                                         copy=False)
+    # Save the sprite
+    imsave(output_sprite, sprite, vmin=vmin, vmax=vmax, cmap=cmap,
+           format=format)
+
+    # Save the parameters
+    if type(vmin).__module__ == 'numpy':
+        vmin = vmin.tolist()  # json does not deal with numpy array
+    if type(vmax).__module__ == 'numpy':
+        vmax = vmax.tolist()  # json does not deal with numpy array
+
+    if output_json is not None:
+        params = {
+                    'nbSlice': {
+                        'X': data.shape[0],
+                        'Y': data.shape[1],
+                        'Z': data.shape[2]
+                    },
+                    'min': vmin,
+                    'max': vmax,
+                    'affine': img.affine.tolist()
+                 }
+        if isinstance(output_json, str):
+            f = open(output_json, 'w')
+            f.write(json.dumps(params))
+            f.close
+        else:
+            output_json.write(json.dumps(params))
+
+    # save the colormap
+    if output_cmap is not None:
+        data = np.arange(0, n_colors)/(n_colors-1)
+        data = data.reshape([1, n_colors])
+        imsave(output_cmap, data, cmap=cmap, format=format)
+
+    return sprite
+
+
+def _custom_cmap(cmap, vmin, vmax, threshold=None):
+    # For internal use only
```

Comment:
please see if you cannot use js_plotting_utils.colorscale or some other nilearn function to avoid duplication

Reply:
I don't think that particular snippet exists as an independent function. 
It is taken from nilearn/plotting/displays.py#L754-L772 
The right move would be to create a small private function in displays.py, and then simply re-use in html_stat_map. 
I was planning to deal with this in a separate PR.

Reply:
i agree that would be the best solution. the created function could probably be used in colorscale too

Reply:
I have implemented a private function called _threshold_cmap in cm.py

Reply:
I also simplified the code of display.py to use this new function.